Document/0x05b-Basic-Security_Testing.md

```diff
@@ -41,29 +41,50 @@ Note: On Linux, you need to choose an SDK directory. `/opt`, `/srv`, and `/usr/l
+
 ##### Testing on a Real Device
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1226>
+Almost any physical device can be used for testing, but there are a few considerations to be made. First, the device needs to be rootable. This is not always the case, and some devices may only be unlocked once the carrier contract has been terminated.
+
+The best candidates are flagship Google devices, since all required resources (bootloader, radio, firmware, ...) can be found online, and you can build your own OS from the official sources if you want to. The devices are always popular in the (rooting) community and you will have much support if anything goes wrong. Finally, they have one of the longest support windows of any available devices, which allows you to use the latest Android versions for a few years.
 
-For dynamic analysis, you'll need an Android device to run the target app on. In principle, you can test without a real Android device and use only the emulator. However, apps execute quite slowly on the emulator, and this can make security testing tedious. Testing on a real device makes for a smoother process and a more realistic environment.
+Devices that are supported by the [LineageOS](https://lineageos.org/ "LineageOS") project are also very good candidates for test devices. They have an active community, easy to follow flashing and rooting instructions and the latest Android versions are typically quickly available as a Lineage installation. LineageOS also continues support for new Android versions long after the OEM has stopped distributing updates.
 
 When working with an Android physical device, you'll want to enable Developer Mode and USB debugging on the device in order to use the ADB debugging interface. Since Android 4.2, the "Developer options" sub menu in the Settings app is hidden by default. To activate it, tap the "Build number" section of the "About phone" view seven times. Note that the build number field's location varies slightly by device—for example, on LG Phones, it is under "About phone -> Software information." Once you have done this, "Developer options" will be shown at bottom of the Settings menu. Once developer options are activated, you can enable debugging with the "USB debugging" switch.
 
-##### Testing on the Emulator
+##### Testing on an Emulator
+
+Multiple emulators exist, once again with their own strengths and weaknesses:
+
+- [Android Virtual Device (AVD)](https://developer.android.com/studio/run/managing-avds.html "Create and Manage Virtual Devices") - The official android emulator, distributed with Android Studio.
+- [Android X86](https://www.android-x86.org/ "Android X86") - An x86 port of the Android code base
+- [Genymotion](https://www.genymotion.com/fun-zone/ "Genymotion") - Mature emulator with many features. Free version available for non-commercial use.
```

Comment:
Can we add: Correlium ?

Reply:
I've added it, but this is the Android section and I don't think they've already officially announced their Android support yet.